purchase_sale_link/models/sale_order.py

```diff
@@ -0,0 +1,92 @@
+# © 2019 Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import models, api, fields
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    purchase_order_count = fields.Integer(
+        string="Number of Purchase Order",
+        compute='_compute_purchase_order_count',
+        groups='purchase.group_purchase_user',
+    )
+
+    @api.multi
+    def get_purchase_orders_service(self):
+        """Get the Purchase Orders linked to one Sale Order
+        by service products purchased automatically"""
+        self.ensure_one()
+
+        purchase_order_lines_service = self.env['purchase.order.line']\
+            .search([('sale_order_id', 'in', self.ids)])
+
+        purchase_orders_service = \
+            purchase_order_lines_service.mapped('order_id')
+
+        return purchase_orders_service
+
+    @api.multi
+    def get_purchase_orders_stock(self):
+        """Get the Purchase Orders linked to one Sale Order
+        by storable products purchased by 'Make To Order' route"""
+        self.ensure_one()
+
+        # Get the related stock moves via the SO's procurement_group_id
+        procurement_group_id = self.procurement_group_id.id
+        stock_moves = self.env['stock.move'].search([
+            ('group_id', '=', procurement_group_id),
+            ('created_purchase_line_id', '!=', 'False'),
+        ])
```

Comment:
When including draft and sent sale orders there should also be a check if procurement group is set. Currently the method also returns purchase orders with no procurement group set.

Reply:
When this change is applied get_purchase_orders_stock should also take into account that procurement group is actually set.